### PR TITLE
Mode xpansion - splitting MPS files depending on simulation week

### PR DESCRIPTION
**Pull request by guilpier-code**
In mode expansion, the first simulation week prints the fixed part of its optimization problem, and each week (first week included) prints te variable part of the simulation.
Thus the first week of simulation has to be spotted by the code.

This PR "merges" the 2 foolowing PRs : #124 and #110

### CODE REVIEW

**Comment by guilpier-code** [comment]
**Questions on this issue** :
- Currently,  the expansion mode is a particular value of the parameter **simulation mode**. This parameter can take the following values : ecnonmy, adequacy, adequcy-draft and expansion (expansion cannot be set via GUI, it can only be set manually, in the **generaldata.ini file**). For the current work, we made the printings of new split MPS files relie on the value the paramater  **simulation mode**. It is set to Expansion then we split MPS into fixed and variable parts. Is this what we want ?
- Currently, in the expansion mode, user do not need to ask for printing special split MPS files, it is done automatically. Should the user be asking for exporting MPS in the study ? And otherwise (user does not need to ask), how does the : GUI > Configuration > optimization preferences > Export MPS should behave ?

**Comment by flomnes** [comment]
Resolve conflicts

File: src/solver/optimisation/opt_appel_solveur_lineaire.cpp [hunk 1]
```diff
@@ -415,6 +430,236 @@ void OPT_EcrireResultatFonctionObjectiveAuFormatTXT(void* Prob,
...
+            }
+
+            il++;
+        }
+    }
+
+    free(Cder);
+
+    auto& study = *Data::Study::Current::Get();
+    Flot = study.createFileIntoOutputWithExtension("fixed-part", "mps", numSpace);
```
Comment (flomnes) [inline src/solver/optimisation/opt_appel_solveur_lineaire.cpp]:
```suggestion
    Flot = study.createFileIntoOutputWithExtension("problem-fixed-part", "mps", numSpace);
```

File: src/solver/optimisation/opt_appel_solveur_lineaire.cpp [hunk 2]
```diff
@@ -415,6 +430,236 @@ void OPT_EcrireResultatFonctionObjectiveAuFormatTXT(void* Prob,
...
+
+    if (Nombre == NULL)
+    {
+        logs.fatal() << "Not enough memory";
+        AntaresSolverEmergencyShutdown();
+    }
+
+    // gp : to be changed
+    auto& study = *Data::Study::Current::Get();
+    Flot = study.createFileIntoOutputWithExtension("variable-part", "mps", numSpace);
```
Comment (flomnes) [inline src/solver/optimisation/opt_appel_solveur_lineaire.cpp]:
```suggestion
    Flot = study.createFileIntoOutputWithExtension("problem-variable-part", "mps", numSpace);
```

**Pull request merged**